client/filters.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"encoding/json"
+	"net/url"
+)
+
+// Filters describes a predicate for an API request.
+//
+// Each entry in the map is a filter term.
+// Each term is evaluated against the set of values.
+// A filter term is satisfied if any one of the values in the set is a match.
+// An item matches the filters when all terms are satisfied.
+//
+// Like all other map types in Go, the zero value is empty and read-only.
+type Filters map[string]map[string]bool
+
+// Add appends values to the value-set of term.
```

Comment:
Related to the above; do we need to make a mention of it panicking if Filter is nil? The issue with nice methods (like `Add`) makes it easier to assume "it just works" (whereas dealing with a raw `map` makes it more obvious "this must be initialised before use!"

Reply:
I called that out in the documentation for the type.
```go
// Like all other map types in Go, the zero value is empty and read-only.
```

The Go authors did not feel it necessary to document that [("net/url".Values).Add()](https://pkg.go.dev/net/url#Values.Add) [panics on a nil receiver](https://go.dev/play/p/4OSk565nrt-). It's the expected convention that nil-valued map types are not writable.

Reply:
That one definitely tripped me up a few times though 😂